src/frontend/src/binder/select.rs

```diff
@@ -182,6 +182,12 @@ impl Binder {
                     aliases.extend(names);
                 }
                 SelectItem::Wildcard => {
+                    if self.context.range_of.is_empty() || self.context.columns.is_empty() {
+                        return Err(ErrorCode::BindError(
```

Comment:
`self.context.columns.is_empty()` is enough? only `columns` store all the columns, `indices_of` and `range_of` just store indices over `columns`

Reply:
Yes, although I was worried about the edge case where there is a table with no columns, I'm not sure if that's possible and in that case we may want to return no columns and not error...

As a matter of fact, this is allowable in postgres:
```
postgres=# create table y ();
CREATE TABLE
postgres=# select * from y;
--
(0 rows)
```

I can add this as a test. It should not error on column-less table. So I'll remove no columns check.